src/pudl/helpers.py

```diff
@@ -1242,11 +1242,10 @@ def generate_rolling_avg(
         .mean(numeric_only=True)
     )
     # with the aggregated data, get a rolling average
-    roll = bones.rolling(window=window, center=True, **kwargs).agg({data_col: "mean"})
-    # return the merged
-    return bones.merge(
-        roll, on=group_cols + ["report_date"], suffixes=("", "_rolling")
-    ).reset_index()
+    bones[f"{data_col}_rolling"] = bones.groupby(by=group_cols)[data_col].transform(
+        lambda x: x.rolling(window=window, center=True, **kwargs).mean()
+    )
```

Comment:
the source of the problem.... the `rolling` was being done after the groupby above but it was not actually using the `group_cols`. (the two example bad plants had lower cost BIT and higher cost DTO and the BIT records were being filled with the higher cost DTO because those records came right after the BIT records. bad bad).

I swapped the `DataFrame.rolling` for a `groupby.DataFrameGroupBy.rolling`.

Reply:
Uuuugh. 🤦🏼